### PR TITLE
QuantumCircuit.ccx(..., ctrl_state=...) parameter added for consistency with other controlled gates

### LINKED ISSUE #6465: Some controlled gates have ctrl_state parameters. Some not.
### What is the current behavior?

Based on https://github.com/Qiskit/qiskit-terra/issues/5098#issue-705776539

The `cx` method has `ctrl_state`  while `ccx` does not have that parameter.
```
from qiskit import QuantumCircuit

qc = QuantumCircuit(3)
qc.cx(0, 1, ctrl_state='0')
qc.ccx(0, 1, 2, ctrl_state='00')
```
```
TypeError: ccx() got an unexpected keyword argument 'ctrl_state'
```

### What is the expected behavior?

Consistency. Either the all controlled gate should have `ctrl_state` parameter  (vote with 🎉 ) or neither (vote with 🚀).

### PR DESCRIPTION
**Pull request by singhmeet11**
### Summary
Fixes #6465 

### Details and comments
**Issue -**  `ctrl_state` was not recognized by ccx gate.
**Problem -** `quantumcircuit.py` did not take `ctrl_state` as a parameter whereas `x.py` in standard gates already had that parameter. \
And as `quantumcircuit.py` imports the circuit from `x.py` in standard gates, there is just a small missing part in `quantumcircuit.py`. 
**Fix -**  Added the parameter `ctrl_state` in `quantumcircuit.py`.
\
**Addition in Test**
Added a test to the `qiskit-terra/test/python/circuit/test_controlled_gate.py`, as it was not able to detect that `ccx` was not able to identify the missing of parameter `ctrl_state`.\
The new test will be able to detect any parameter inconsistency between `ccx()` gate in `quantumcircuit.py` and standard gates.  

Also corrected a small typo , cxx -> ccx

### CODE REVIEW

**Comment by iamsantanubanerjee** [comment]
I was taking a look at this too. Do you think we should also add the 'label' parameter as well? Also, I see a little inconsistency with the formatting of your code. Do you think we can collaborate to make a better PR?

**Comment by iamsantanubanerjee** [comment]
@singhmeet11 Can you please tell me how did you pass all the checks and what was the issue before?

**Comment by singhmeet11** [comment]
@iamsantanubanerjee  So there were two main issues: 
1. One of the lines were too long and one was dividing the square bracket into two parts. So what I did was looked at the syntax of cx and one other gate above(I do not remember which one) and made changes accordingly. It took me a few tries tho.
2. There was some problem from the backend so when I updated my PR with the new commits from qiskit-terra it cleared all the tests.

**Review by javabster — COMMENTED** [review]
This is looking good @singhmeet11 😄 Could you add a test as well please? You could use the example from the description in the original issue as a starting point

**Comment by singhmeet11** [comment]
Thanks for reviewing the PR @javabster ! Could you please guide me a bit about how do I create custom tests. I did not know that we had to do that, and no need to go into detail just give me some reference which talks about this. And if it is possible could you also tell me if I can execute the test on Github only, because I deleted the repository due to some space issues on my laptop, otherwise I will have to go through the whole procedure again.

**Comment by javabster** [comment]
Hi @singhmeet11 😄 you can take a look [here](https://github.com/Qiskit/qiskit-terra/blob/main/CONTRIBUTING.md#test) for some general guidance on running the tests (you will need to do this on your own machine). For this particular case I'd recommend adding a new test to `qiskit-terra/test/python/circuit/test_controlled_gate.py`, you can look at the other tests in that file for guidance on how to structure tests. Also in the issue #6455 description there is an example of what the expected functionality is, this could be a good start for a test case.

For information on requirements for pull requests you can check out the PR checklist [here](https://github.com/Qiskit/qiskit-terra/blob/main/CONTRIBUTING.md#pull-request-checklist)

**Comment by singhmeet11** [comment]
@javabster  I made changes in the `qiskit-terra/test/python/circuit/test_controlled_gate.py` (`test_open_control_ccx_unrolling` to be more accurate).  How does it look ??

**Comment by javabster** [comment]
Hi @singhmeet11 I think the previous `test_open_control_ccx_unrolling` is still a valid implementation, so instead changing it I think it would be best to leave it as it is and add a new test that covers the new functionality you've implemented

**Comment by singhmeet11** [comment]
@javabster let me know if the test looks okay?? 
I sort of ran into a problem so that is the reason for so many commits, `tox -eblack` was passing on my laptop but was failing on github(no idea why) and after being stuck for a day @woodsp-ibm helped me out, thanks again.

File: test/python/circuit/test_controlled_gate.py [hunk 1]
```diff
@@ -884,6 +884,17 @@ def test_open_control_cxx_unrolling(self):
         ref_dag = circuit_to_dag(ref_circuit)
         self.assertEqual(unrolled_dag, ref_dag)
 
+    def test_ccx_parameter_consistency(self):
+        """Test the consistency of parameters"""
```
Comment (javabster) [inline test/python/circuit/test_controlled_gate.py:888]:
```suggestion
        """Test the consistency of parameters
        See issue: https://github.com/Qiskit/qiskit-terra/issues/6465
        """
```

**Review by javabster — DISMISSED** [review]
Just one small suggestion for mentioning the issue in the test, otherwise LGTM 🚀

**Comment by singhmeet11** [comment]
@javabster Done

**Review by javabster — APPROVED** [review]
LGTM 🚀 

@1ucian0 thoughts?

File: test/python/circuit/test_controlled_gate.py [hunk 2]
```diff
@@ -884,6 +884,19 @@ def test_open_control_cxx_unrolling(self):
         ref_dag = circuit_to_dag(ref_circuit)
         self.assertEqual(unrolled_dag, ref_dag)
 
+    def test_ccx_parameter_consistency(self):
+        """Test the consistency of parameters
```
Comment (1ucian0) [inline test/python/circuit/test_controlled_gate.py:888]:
```suggestion
    def test_ccx_ctrl_state_consistency(self):
        """Test the consistency of parameters ctrl_state in CCX
```

**Comment by 1ucian0** [comment]
Test renamed. Title renamed and on its way to merge! Thanks!

**Review by 1ucian0 — APPROVED** [review]

**Pull request merged**